cirq/protocols/resolve_parameters.py

```diff
@@ -88,10 +89,13 @@ def resolve_parameters(
     param_resolver = ParamResolver(param_resolver)
     if isinstance(val, sympy.Basic):
         return param_resolver.value_of(val)
+    if isinstance(val, Sequence):
+        return type(val)(resolve_parameters(e, param_resolver) for e in val)
```

Comment:
This is not safe. Sequence types are not guaranteed to have a constructor that takes a generator. I think all a type has to do to be a sequence is have `__len__` and `__getitem__`.

I'd recommend having an explicit list of types we support, like just tuple and list and dict.

Reply:
Okay, I changed it to list and tuples explicitly.

Is it clear to you how dicts should be resolved (i.e., just the keys, just the values, or both)? I think it's ambiguous enough that we should wait for a use case.

Reply:
I agree.